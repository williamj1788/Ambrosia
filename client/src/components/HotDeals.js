import React from 'react';
import s from '../styles/HotDeals.module.scss';
import PizzaIcon from '../images/Pizza_icon_white.png';
import { connect } from 'react-redux';

export class HotDeals extends React.Component{

    findProductsWithDiscounts = () => {
        return this.props.products.filter(product => {
            return product.discountObj.length !== 0;
        })
    }
    
    render(){
        return(
            <section className={s.hotDeals}>
                <h3 className={s.hotDealsTitle}>Hot Deals</h3>
                <DealContainer products={this.findProductsWithDiscounts()} />
            </section>
        )
    }
}

export const DealContainer = ({ products }) => {
    const deals = products.map(product => {
        return <Deal {...product} />
    });


    return(
        <div className={s.dealsContainer}>
            {deals}
        </div>
    )
}


export const Deal = ({ name, picture, description, price, discountObj }) => {
    return(
        <div className={s.deal}>
            <img className={s.dealImg} src={picture} alt="product"/>
            <div className={s.dealInfo}>
                <span className={s.dealTitle}>{name}</span>
                <p className={s.dealDesc}>{description}</p>
                <div className={s.dealStat}>
                    <span className={s.dealPrice}>{getDiscountPercent(discountObj[0].price, price) + '% off'}</span>
                    <span className={s.dealPrice}>{Math.floor(1 + daysBetweenDates(discountObj[0].expiresAt, new Date())) + ' days left'}</span>
                </div>
                <div className={s.dealPriceContainer}>
                    <span className={s.dealPrice}><s>{'$' + price}</s></span>
                    <span className={s.dealPrice}>{'$' + discountObj[0].price}</span>
                    <button className={s.dealButton}>Place Order</button>
                </div>
            </div>
        </div>
    )
}

function roundNumber(num, target = 2){
    return Math.round(num * Math.pow(10, target)) / 100
};

function getDiscountPercent(discount, price) {
   return Math.round((1 - roundNumber(discount / price)) * 100)
}

function daysBetweenDates(day1, day2) {
    console.log(day1, day2)
    console.log(Math.round(((new Date(day1).getTime() - day2.getTime()) / (1000 * 60 * 60 * 24)) * 100) / 100)
    return Math.round(((new Date(day1).getTime() - day2.getTime()) / (1000 * 60 * 60 * 24)) * 100) / 100
}

export default connect(state => {
    return {
        products: state.products
    }
})(HotDeals);