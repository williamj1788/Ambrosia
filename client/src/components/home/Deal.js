import React from 'react';
import s from '../../styles/HotDeals.module.scss';
import { addOrder, removeOrder } from '../../redux/action';
import uuid from 'uuid';
import { connect } from 'react-redux';
class Deal extends React.Component{
    
    handleClick = () => {
        const { _id, name, price, discountObj } = this.props;
        const order = {
            id: uuid(),
            productID: _id, 
            name, 
            price: discountObj.length ? discountObj[0].price : price, 
            qty: 1
        };
        this.addOrder(order);
    }
    
    addOrder = order => {
        this.props.dispatch(addOrder(order));
    }

    removeOrder = id => {
        this.props.dispatch(removeOrder(id))
    }
    
    render(){
        const { _id, name, picture, description, price, discountObj } = this.props;
        const order = this.props.orders.find(x => x.productID === _id);
        return(
            <div className={s.deal}>
                <img className={s.dealImg} src={picture} alt="product"/>
                <div className={s.dealInfo}>
                    <span className={s.dealTitle}>{name}</span>
                    <p className={s.dealDesc}>{description}</p>
                    <div className={s.dealPriceContainer}>
                        <span className={s.dealPrice}>{getDiscountPercent(discountObj[0].price, price) + '% off'}</span>
                        <span className={s.dealPrice}><s>{'$' + price.toFixed(2)}</s></span>
                        <span className={s.dealPrice}>{'$' + discountObj[0].price.toFixed(2)}</span>
                        {order 
                        ? <button onClick={() => this.removeOrder(order.id)} className={s.cartButton}>{`In Cart: ${order.qty}`}</button>
                        : <button onClick={this.handleClick} className={s.dealButton}>Place Order</button>}
                    </div>
                </div>
            </div>
        )
    }
}

function roundNumber(num, target = 2){
    return Math.round(num * Math.pow(10, target)) / 100
};

function getDiscountPercent(discount, price) {
   return Math.round((1 - roundNumber(discount / price)) * 100)
}

export default connect(state => {
    return{
        orders: state.orders
    }
})(Deal);
