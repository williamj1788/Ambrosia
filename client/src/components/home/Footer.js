import React from 'react';
import s from '../../styles/Footer.module.scss';
import Title from '../shared/Title';

import locationIcon from '../../images/location-icon.png';
import phoneIcon from '../../images/phone-icon.png';
import emailIcon from '../../images/email-icon.svg';

export const Footer = () => {
    return(
        <footer className={s.footer}>
            <div className={s.container}>
                <Opening />
                <Contact />
            </div>
        </footer> 
    )
}

export const Opening = () => {
    return(
        <div className={s.opening}>
            <h4 className={s.title}>Opening Hours</h4>
            <div className={s.hours}>
                <HoursSection date={'Monday - Friday'} time={'8.00 am - 10.00 pm'} />
                <HoursSection date={'Saturday'} time={'10.00 am - 5.00 pm'} />
                <HoursSection date={'Sunday'} time={'Closed'} />
            </div>
        </div>
    )
}

export const HoursSection = ({ date, time }) => {
    return(
        <div className={s.hoursSection}>
            <p className={s.hoursTitle}>{date}</p>
            <p className={s.hoursText}>{time}</p>
        </div>
    )
}

export const Contact = () => {
    return(
        <div className={s.contact}>
            <h4 className={s.title}>Contact Us</h4>
            <div className={s.contactSocialIconContainer}>
                <LinkIcon href='https://www.twitter.com' className={s.twitterIcon} />
                <LinkIcon href='https://www.facebook.com' className={s.facebookIcon} />
                <LinkIcon href='https://www.instagram.com' className={s.instagramIcon} />
            </div>
            <ContactSection img={locationIcon} text={'203 Fake St. Mountain View, San Francisco, California, USA'} />
            <ContactSection img={phoneIcon} text={'+1 978-236-7889'} />
            <ContactSection img={emailIcon} text={'JacquezWilliams@teleworm.us'} />
        </div>
    )
}

export const LinkIcon = ({ href, className }) => {
    return(
        <a href={href} target='_blank' rel="noopener noreferrer" className={`${s.contactSocialIcon} ${className}`} ></a>
    )
}

export const ContactSection = ({ img, text }) => {
    return(
        <div className={s.contactSection}>
            <img className={s.contactSectionIcon} src={img} alt="Icon"/>
            <p className={s.contactSectionText}>{text}</p>
        </div>
    )
}

export default Footer;