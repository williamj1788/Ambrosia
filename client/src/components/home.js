import React from 'react';

import Navbar from './Navbar';
import ReviewSection from './ReviewSection';
import Footer from './Footer';
import HotDeals from './HotDeals';

import s from '../styles/Home.module.scss';

import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

import PizzaIcon from '../images/Pizza_icon_white.png';
import MoneyIcon from '../images/Money_icon.png';
import TruckIcon from '../images/truck_icon.png';
import ArrowDown from '../images/arrow-down.png';


export class Home extends React.Component{
    
    state = {
        redirectToMenu: false
    }

    setRedirect = () => {
        this.setState({
            redirectToMenu: true,
        });
    }
    
    render(){
        if(this.state.redirectToMenu){ 
            return <Redirect to='/menu' />
        }
        return(
            <div>
                <Hero redirect={this.setRedirect}/>
                <About />
                <HotDeals />
                <ReviewSection />
                <Footer />
                <Copyright />
            </div>
        )
    }
}

const Hero = ({ redirect }) => {
    return(
        <section className={s.hero}>
            <div className={s.darken}>
                <Navbar />
                <p className={s.heroTitle}>Best Pizza in Town</p>
                <button onClick={redirect} className={s.heroButton}>View Menu</button>
                <img className={s.heroArrow} src={ArrowDown} alt="ArrowDown Icon"/>
            </div>
        </section>
    )
}

const About = () => {
    return(
        <section className={s.about}>
            <AboutTitle />
            <div className={s.flexContainer} >
                <Unit 
                img={PizzaIcon}
                title={'Great Pizza'}
                desc={'We go through multiple checks of each ingredient to ensure that our pizza is top quality'}
                />
                <Unit 
                img={MoneyIcon}
                title={'Low Prices'}
                desc={'We work with our local farmers to provide affordable and quality ingredients and pass the savings to our customers'}
                />
                <Unit 
                img={TruckIcon}
                title={'Fast Service'}
                desc={'With active drivers that are always on standby, we guarantee a delivery in under 30 mins or the order is free'}
                />
            </div>
        </section>
    )
}

const AboutTitle = () => {
    return(
        <div>
            <h2 className={s.aboutTitle}>About Us</h2>
            <p className={s.aboutDesc}>We are committed to providing our customers with excellent service that makes us deserving of being the best pizzeria in town</p> 
        </div>
    )
}

export const Unit = ({ img, title, desc }) => {
    return(
        <div className={s.AboutUnits}>
            <img className={s.aboutUnitImg} src={img} alt="Icon"/>
            <span className={s.aboutUnitTitle} >{title}</span>
            <p className={s.aboutUnitDesc}>{desc}</p>
        </div>
    )
}

const Copyright = () => {
    return(
        <div className={s.copyright}>
            <p className={s.copyrightText}>Copyright Â©2019 All rights reserved</p>
        </div>
    )
}

export default connect()(Home);