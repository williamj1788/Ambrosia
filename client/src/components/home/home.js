import React from 'react';

import Navbar from '../shared/Navbar';
import ReviewSection from './ReviewSection';
import Footer from './Footer';
import HotDeals from './HotDeals';

import s from '../../styles/Home.module.scss';

import { connect } from 'react-redux';
import { setProducts } from '../../redux/action';
import { Redirect } from 'react-router-dom';


import PizzaIcon from '../../images/Pizza_icon_white.png';
import MoneyIcon from '../../images/Money_icon.png';
import TruckIcon from '../../images/truck_icon.png';
import ArrowDown from '../../images/arrow-down.png';


export class Home extends React.Component{
    
    state = {
        redirectToMenu: false,
        loading: !this.props.products
    }

    componentDidMount(){
        if(!this.props.products){
            this.loadProducts();
        }
    }

    loadProducts = () => {
        this.fetchProducts()
        .then(products => this.props.dispatch(setProducts(products)))
        .then(() => this.setState({loading: false}))
    }

    fetchProducts = () => {
       return fetch('/api/admin/products')
        .then(res => res.json())
        .catch(console.log);
    }

    setRedirect = () => {
        this.setState({
            redirectToMenu: true,
        });
    }
    render(){
        if(this.state.redirectToMenu){ 
            return <Redirect push to='/menu' />
        }
        if(this.state.loading){
            return <div className='loading'>Loading...</div>
        }
        return(
            <main>
                <Hero redirect={this.setRedirect}/>
                <About />
                <HotDeals />
                <ReviewSection />
                <Footer />
                <Copyright />
            </main>
        )
    }
}

const Hero = ({ redirect }) => {
    return(
        <section className={s.hero}>
            <div className={s.darken}>
                <Navbar />
                <p className={s.heroTitle}>Best Pizza in Town</p>
                <button onClick={redirect} className={s.heroButton}>View Menu</button>
                <img className={s.heroArrow} src={ArrowDown} alt="ArrowDown Icon"/>
            </div>
        </section>
    )
}

const About = () => {
    return(
        <section className={s.about}>
            <AboutTitle />
            <div className={s.flexContainer} >
                <Perk 
                img={PizzaIcon}
                title={'Great Pizza'}
                desc={'We go through multiple checks of each ingredient to ensure that our pizza is top quality'}
                />
                <Perk 
                img={MoneyIcon}
                title={'Low Prices'}
                desc={'We work with our local farmers to provide affordable and quality ingredients and pass the savings to our customers'}
                />
                <Perk 
                img={TruckIcon}
                title={'Fast Service'}
                desc={'With active drivers that are always on standby, we guarantee a delivery in under 30 mins or the order is free'}
                />
            </div>
        </section>
    )
}

const AboutTitle = () => {
    return(
        <div>
            <h2 className={s.aboutTitle}>About Us</h2>
            <p className={s.aboutDesc}>We are committed to providing our customers with excellent service that makes us deserving of being the best pizzeria in town</p> 
        </div>
    )
}

export const Perk = ({ img, title, desc }) => {
    return(
        <div className={s.AboutUnits}>
            <img className={s.aboutUnitImg} src={img} alt="Icon"/>
            <span className={s.aboutUnitTitle} >{title}</span>
            <p className={s.aboutUnitDesc}>{desc}</p>
        </div>
    )
}

const Copyright = () => {
    return(
        <div className={s.copyright}>
            <p className={s.copyrightText}>Copyright Â©2019 All rights reserved</p>
        </div>
    )
}

export default connect(state => {
    return {
        products: state.products
    }
})(Home);