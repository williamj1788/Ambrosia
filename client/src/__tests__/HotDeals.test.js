import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HotDeals, DealContainer, Deal } from "../components/home/HotDeals";

describe('<HotDeals />', () => {
    test('Snapshot Test', () => {
        const component = shallow(<HotDeals />);
        expect(toJson(component)).toMatchSnapshot();
    });
});

describe('<DealContainer />', () => {
    test('Snapshot Test', () => {
        const component = shallow(<DealContainer />);
        expect(toJson(component)).toMatchSnapshot();
    });
});

describe('<Deal />', () => {
    test('Snapshot Test', () => {
        const component = shallow(<Deal />);
        expect(toJson(component)).toMatchSnapshot();
    });
});