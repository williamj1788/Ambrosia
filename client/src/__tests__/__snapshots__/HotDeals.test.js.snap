// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Deal /> Snapshot Test 1`] = `
<div
  className="deal"
>
  <img
    alt=""
    className="dealImg"
    src="Pizza_icon_white.png"
  />
  <div
    className="dealInfo"
  >
    <span
      className="dealTitle"
    >
      Pizza Title
    </span>
    <p
      className="dealDesc"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
    </p>
    <div
      className="dealPriceContainer"
    >
      <span
        className="dealPrice"
      >
        <s>
          10.99
        </s>
      </span>
      <span
        className="dealPrice"
      >
        7.99
      </span>
      <button
        className="dealButton"
      >
        Place Order
      </button>
    </div>
  </div>
</div>
`;

exports[`<DealContainer /> Snapshot Test 1`] = `
<div
  className="dealsContainer"
>
  <Deal />
  <Deal />
  <Deal />
  <Deal />
  <Deal />
  <Deal />
  <Deal />
  <Deal />
  <Deal />
  <Deal />
</div>
`;

exports[`<HotDeals /> Snapshot Test 1`] = `
<section
  className="hotDeals"
>
  <h3
    className="hotDealsTitle"
  >
    Hot Deals
  </h3>
  <DealContainer />
</section>
`;
